### DISCUSSION TITLE
Drop `controller-runtime` dependency

### DISCUSSION THREAD

**Discussion by maiqueb** [k8snetworkplumbingwg/whereabouts#205] [Ideas]
Hey,

I've started not long ago a prototype of a reconcile loop for IP addresses for whereabouts `IPPool`s in PR #185 . The approach I'm using just uses the informer / lister / clientSet Kubernetes library objects, which I'm introducing in PR #202 .

If we add these library objects to the github repo, we can drop the controller-runtime dependency - I've ran a couple of times already into some vendoring related issues around it. 

Another option would be to - instead of removing the dependency - use more if it, and manage the informers / cache objects through its framework, rather than via these library objects.

The former option seems more in line with the `k8snetworkplumbingwg` projects, since it is used (at least) in:
  - https://github.com/k8snetworkplumbingwg/network-resources-injector
  - https://github.com/k8snetworkplumbingwg/multi-networkpolicy-iptables

**Comment by dougbtv** [comment]
I'm in favor of trying to have some normalization across the NPWG (`k8snetworkplumbingwg`) projects. So, if we think that removing the dependency might reduce some of our woes.

I'm also hopeful that we can build the ip-reconciler controller as a kind of model that we can use as a reference when working through a controller for the Multus dynamic attachments controller.

However, if it's not practical, all bets are off -- I'm ok to double down on using it.

**Reply by s1061123** [reply]
I'm second on that. Less dependency library improves maintenability, such as update vendor. So if possible, I prefer to remove controller-runtime.

**Comment by maiqueb** [comment]
Implemented in #225 .

**Reply by maiqueb** [reply]
... which was merged.
